tests/integration/adapter/incremental/test_distributed_incremental.py

```diff
@@ -0,0 +1,205 @@
+
+
+lw_delete_schema = """
+version: 2
+
+models:
+  - name: "lw_delete_inc"
+    description: "Incremental table"
+"""
+
+lw_delete_inc = """
+{{ config(
+        materialized='distributed_incremental',
+        order_by=['key1'],
+        unique_key='key1',
+        incremental_strategy='delete+insert'
+    )
+}}
+{% if is_incremental() %}
+   WITH (SELECT max(key1) - 20 FROM lw_delete_inc) as old_max
+   SELECT assumeNotNull(toUInt64(number + old_max + 1)) as key1, toInt64(-(number + old_max)) as key2, toString(number + 30) as value FROM numbers(100)
+{% else %}
+   SELECT toUInt64(number) as key1, toInt64(-number) as key2, toString(number) as value FROM numbers(100)
+{% endif %}
+"""
+
+
+class TestLWDeleteDistributedIncremental:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"lw_delete_inc.sql": lw_delete_inc}
+
+    @pytest.mark.skipif(
+        os.environ.get('DBT_CH_TEST_CLUSTER', '').strip() == '', reason='Not on a cluster'
+    )
+    def test_lw_delete(self, project):
+        run_dbt()
+        result = project.run_sql("select count(*) as num_rows from lw_delete_inc", fetch="one")
+        assert result[0] == 100
+        run_dbt()
+        result = project.run_sql("select count(*) as num_rows from lw_delete_inc", fetch="one")
+        assert result[0] == 180
+
+
+compound_key_schema = """
+version: 2
+
+models:
+  - name: "compound_key_inc"
+    description: "Incremental table"
+"""
+
+compound_key_inc = """
+{{ config(
+        materialized='distributed_incremental',
+        order_by=['key1', 'key2'],
+        unique_key='key1, key2',
+        incremental_strategy='delete+insert'
+    )
+}}
+{% if is_incremental() %}
+   WITH (SELECT max(key1) - 20 FROM compound_key_inc) as old_max
+   SELECT assumeNotNull(toUInt64(number + old_max + 1)) as key1, toInt64(-key1) as key2, toString(number + 30) as value FROM numbers(100)
+{% else %}
+   SELECT toUInt64(number) as key1, toInt64(-number) as key2, toString(number) as value FROM numbers(100)
+{% endif %}
+"""
+
+
+class TestDistributedIncrementalCompoundKey:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"compound_key_inc.sql": compound_key_inc}
+
+    @pytest.mark.skipif(
+        os.environ.get('DBT_CH_TEST_CLUSTER', '').strip() == '', reason='Not on a cluster'
+    )
+    def test_compound_key(self, project):
+        run_dbt()
+        result = project.run_sql("select count(*) as num_rows from compound_key_inc", fetch="one")
+        assert result[0] == 100
+        run_dbt()
+        result = project.run_sql("select count(*) as num_rows from compound_key_inc", fetch="one")
+        assert result[0] == 180
+
+
+replicated_seed_schema_yml = """
+version: 2
+
+seeds:
+  - name: base
+    config:
+      engine: ReplicatedMergeTree('/clickhouse/tables/{uuid}/one_shard', '{server_index}' )
+  - name: added
+    config:
+      engine: ReplicatedMergeTree('/clickhouse/tables/{uuid}/one_shard', '{server_index}' )
```

Comment:
Maybe it will be better if test at least once with _**multiple shards** x **multiple replicas**_?
e.g. `ReplicatedMergeTree('/clickhouse/tables/{shard}/{database}/table_name', '{replica}')` (example from https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replication)

Reply:
for integration tests, if we managed to set a multiple shards * multple replicas matrix, data will always exist in one shard (same zk path), maybe it is not worth adding another test?

Reply:
For me it is an usual usage.
But it is more managed by **dbt-core** & the **clickhouse cluster**, so any integration test on the _cluster_ mode is enough.

Reply:
BTW, I did not get the reason why the data persist in one shard.
Do you use `drop table distributed_table_name on cluster cluster_name` or `create or replace table`, or something similar?

Reply:
I meant for the Replicated Engine tables, not distributed tables. If I understand it correctly, during our tests, `insert into` query will be performed by one node and data will be replicated among nodes with the same shard (same zk path).

Reply:
I am not sure if I understand you correctly. 
`Seeds` in test `TestInsertsOnlyDistributedIncrementalMaterialization` seem to serve the purpose of providing source data and seed function is already being tested in `test_basic.py` but with table materialization only. 

My quetstion is, should we add another test for seed with distributed materialization and check data distribution or we can add another test like `TestDistributedMaterializations` in `test_basic.py` with a different cluster (maybe 1*2 + 1*1) setting?

Reply:
My approach is to make sure seed are using the same shard and different replica, i.e. to broadcast the whole dataset.
I have never used File Table Engine before, could you explain a bit more? How it would help in this situation?

Reply:
`File Table Engine` reads the csv files directly, not sure if it will help.
https://clickhouse.com/docs/en/engines/table-engines/special/file

Reply:
I think now I understand how you test the `distributed_table`. I still recommend a separate `.py` for each main materialization, or, if not applicable, document more.